docs/adr/0003-swift-packaging.md

```diff
@@ -0,0 +1,225 @@
+# Distributing Swift Packages
+
+* Status: proposed
+* Deciders: rfkelly
+* Date: 2021-06-25
+
+## Context and Problem Statement
+
+Our iOS consumers currently obtain application-services as a pre-compiled `.framework` bundle
+distributed via [Carthage](https://github.com/Carthage/Carthage). The current setup is not
+compatible with building on new M1 Apple Silicon machines and has a number of other problems.
+As part of a broader effort to modernize the build process of iOS applications at Mozilla,
+we have been asked to re-evaluate how application-services components are dsitributed for iOS.
+
+See [Problems with the current setup](#problems-with-the-current-setup) for more details.
+
+## Decision Drivers
+
+* Ease-of-use for iOS consumers.
+* Compatibility with M1 Apple Silicon machines.
+* Consistency with other iOS components being developed at Mozilla.
+* Ability for the Nimbus Swift bindings to easily depend on Glean.
+* Ease of maintainability for application-services developers.
+
+## Considered Options
+
+* **(A) Do Nothing**
+  * Keep our current build and distribution setup as-is.
+* **(B) Use Carthage to build XCFramework bundles**
+  * Make a minimal change to our Carthage setup so that it builds
+    the newer XCFramework format, which can support M1 Apple Silicon.
+* **(C) Distribute a single pre-compiled Swift Package**
+  * Convert the all-in-one `MozillaAppServices` Carthage build to a similar
+    all-in-one Swift Package, distributed as a binary artifact.
+* **(D) Distribute multiple source-based Swift Package targets, with pre-compiled Rust code**
+  * Split the all-in-one `MozillaAppServices` Carthage build into a separate
+    Swift Package target for each component, with a shared dependency on pre-compiled
+    Rust code as a binary artiact.
+
+## Decision Outcome
+
+Chosen option: **(D) Distribute multiple source-based Swift Packages, with pre-compiled Rust code**.
+
+This option will provide the best long-term consumer experience for iOS developers, and
+has the potential to simplify maintenance for application-services developers after an
+initial investment of effort.
+
+### Positive Consequences
+
+* Swift packages are very convenient to consume in newer versions of XCode.
+* Different iOS apps can choose to import a different subset of the available components,
+  potentiallying helping keep application size down.
```

Comment:
```suggestion
  potentially helping keep application size down.
```

Reply:
While a valid thing here, I think it's probably negligible for the Swift code compared to shipping the prebuilt Rust library that has code of all components anyway.